iroh/src/commands/sync.rs

```diff
@@ -223,6 +285,70 @@ impl DocCommands {
                     println!("{}", fmt_entry(&entry));
                 }
             }
+            Self::Import {
+                doc,
+                author,
+                prefix,
+                path,
+                in_place,
+            } => {
+                let doc = get_doc(iroh, env, doc).await?;
+                let author = env.author(author)?;
+                let mut prefix = prefix.unwrap_or_else(|| String::from(""));
+
+                if prefix.ends_with('/') {
+                    prefix.pop();
+                }
+                let root = canonicalize_path(&path)?.canonicalize()?;
+                let tag = tag_from_file_name(&root)?;
+                let stream = iroh
+                    .blobs
+                    .add_from_path(
+                        root.clone(),
+                        in_place,
+                        SetTagOption::Named(tag.clone()),
+                        WrapOption::NoWrap,
+                    )
+                    .await?;
+                let (updates, progress) = doc.set_hash_streaming(author).await?;
+                let root_prefix = match root.parent() {
+                    Some(p) => p.to_path_buf(),
+                    None => PathBuf::new(),
+                };
+                let (entries, size) =
+                    import_coordinator(doc.id(), root_prefix, prefix, stream, updates, progress)
```

Comment:
probably needs a better name. engine? It takes the entries off of the stream from `iroh.blobs.add_from_path` and sends them down to the doc using the `doc.set_hash_streaming` sink, and then aggregates the results & powers the progress bars.

Reply:
Coordinator seems fine. An engine I would associate with a stateful thing that stays around for some time.